model/scratch/dffml_model_scratch/model/slr.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+
+from dffml.repo import Repo
+from dffml.source.source import Sources
+from dffml.feature import Features
+from dffml.accuracy import Accuracy
+from dffml.model.model import ModelConfig, ModelContext, Model
+from dffml.util.entrypoint import entry_point
+from dffml.util.cli.arg import Arg
+
+
+class SLRConfig(ModelConfig, NamedTuple):
+    predict: str
+    directory: str  
+
+
+class SLRContext(ModelContext):
+
+    def __init__(self, parent, features):
+        super().__init__(parent, features)
+        self.xData = np.array([])
+        self.yData = np.array([])
+        self.features = self.applicable_features(features)
+
+    def get_regression_line(self):
+        return self.parent.saved.get(hashlib.sha384((''.join(sorted(self.features))).encode()).hexdigest(), None)
+
+    def set_regression_line(self, slope, constant, accuracy):
+        self.parent.saved[hashlib.sha384((''.join(sorted(self.features))).encode()).hexdigest()] = (slope, constant, accuracy)
+
+    def applicable_features(self, featuress):
+        usable = []
+        if len(featuress) != 1:
+            raise ValueError("Simple Linear Regression doesn't support features other than 1")
+        for feature in featuress:
+            if feature.dtype() != int and feature.dtype() != float:
+                raise ValueError("Simple Linear Regression only supports int or float feature")
+            if feature.length() != 1:
+                raise ValueError("Simple LR only supports single values (non-matrix / array)")
+            usable.append(feature.NAME)
+        return usable
+
+    async def predict_input(self, x):
+        regression_line = self.get_regression_line()
+        prediction = regression_line[0]*x+regression_line[1]
+        self.logger.debug("Predicted Value of {} {}:".format(self.parent.config.predict, prediction))
+        return prediction
+
+    async def squared_error(self, ys, yline):
+        return sum((ys - yline)**2)
+
+    async def coeff_of_deter(self, ys, regression_line):
+        y_mean_line = [np.mean(ys) for y in ys]
+        squared_error_mean = await self.squared_error(ys, y_mean_line)
+        squared_error_regression = await self.squared_error(ys, regression_line)
+        return 1 - (squared_error_regression/squared_error_mean)
+
+    async def best_fit_line(self):
+        self.logger.debug("Number of input repos: {}".format(len(self.xData)))
+        x = self.xData
+        y = self.yData
+        mean_x = np.mean(self.xData)
+        mean_y = np.mean(self.yData)
+        m = (mean_x*mean_y - np.mean(x*y))/((mean_x**2) - np.mean(x*x))
+        b = mean_y - (m*mean_x)
+        regression_line = [m*x + b for x in x]
+        accuracy = await self.coeff_of_deter(y, regression_line)
+        self.set_regression_line(m, b, accuracy)
+        return (m, b, accuracy)
+
+    async def train(self, sources: Sources):
+        async for repo in sources.with_features(self.features):
```

Comment:
`async for sources.with_features(self.features + [self.parent.config.predict]):`